pkg/eks/nodegroup.go

```diff
@@ -174,6 +174,21 @@ func hasAmazonLinux2Node(nodeGroups []KubeNodeGroup) bool {
 	return false
 }
 
+// HasInstanceType returns whether some node in the group fulfils the type check
+func HasInstanceType(nodeGroup *api.NodeGroup, hasType func(string) bool) bool {
```

Comment:
I think this function would be more reusable if it was in the api package because it is imported from anywhere (and it only needs information that belongs there). If we leave it in the eks package we won't be able to import it while validating the config file, for example.

(the eks package is mainly for making requests to AWS)